// @vestigefi/widgets/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];
      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }
    return target;
  };
  return _extends.apply(this, arguments);
}

var Chart = function Chart(_ref) {
  var assetId = _ref.assetId,
    poolId = _ref.poolId,
    currency = _ref.currency,
    invert = _ref.invert,
    adjust = _ref.adjust,
    tools = _ref.tools,
    interval = _ref.interval,
    width = _ref.width,
    height = _ref.height,
    className = _ref.className,
    style = _ref.style;
  var src = React.useMemo(function () {
    var url = new URL("https://vestige.fi/widget/" + (assetId || 700965019) + "/chart");
    if (poolId) url.searchParams.append('poolId', poolId.toString());
    if (interval) url.searchParams.append('interval', interval);
    if (currency) url.searchParams.append('currency', currency);
    if (invert !== undefined) url.searchParams.append('invert', invert ? 'true' : 'false');
    if (adjust !== undefined) url.searchParams.append('adjust', adjust ? 'true' : 'false');
    if (tools !== undefined) url.searchParams.append('tools', tools ? 'true' : 'false');
    return url.href;
  }, [assetId, poolId, currency, invert, adjust, tools, interval]);
  return React__default.createElement("iframe", {
    title: 'Vestige.fi Chart',
    src: src,
    className: className,
    style: _extends({
      width: width || 1024,
      height: height || 650
    }, style || {})
  });
};

var Swap = function Swap(_ref) {
  var assetIn = _ref.assetIn,
    assetOut = _ref.assetOut,
    amountIn = _ref.amountIn,
    amountOut = _ref.amountOut,
    width = _ref.width,
    height = _ref.height,
    className = _ref.className,
    style = _ref.style;
  var src = React.useMemo(function () {
    var url = new URL('https://vestige.fi/widget/swap');
    if (typeof assetIn === 'number') url.searchParams.append('asset_in', assetIn.toString());
    if (typeof assetOut === 'number') url.searchParams.append('asset_out', assetOut.toString());
    if (amountIn && typeof amountIn === 'number') url.searchParams.append('amount_in', amountIn.toString());
    if (amountIn && typeof amountOut === 'number') url.searchParams.append('amount_out', amountOut.toString());
    return url.href;
  }, []);
  return React__default.createElement("iframe", {
    title: 'Vestige Swap',
    src: src,
    className: className,
    style: _extends({
      width: width || 400,
      height: height || 450
    }, style || {})
  });
};

exports.Chart = Chart;
exports.Swap = Swap;
